### PR TITLE
add websocket support for server

### PR DESCRIPTION
**Pull request by mathieu-pousse**
Signed-off-by: Mathieu POUSSE <mathieu.pousse@zenika.com>

### CODE REVIEW

File: server/hub.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	game    game.Game
+}
+
+func NewGameHub(game game.Game) *GameHub {
+	hub := GameHub{game: game, clients: make([]*websocket.Conn, 0)}
+	return &hub
+}
+
+func (gh *GameHub) handle(connection *websocket.Conn) {
+	log.Warnf("new ws connection for this fucker")
```
Comment (ggerbaud) [inline server/hub.go]: hum...

File: server/server.go [hunk 2]
```diff
@@ -35,8 +39,10 @@ func (server *Server) Start() {
 	apiRouter.GET("/styles", server.getStylesHandler) // retourne la liste des styles
 	// creation du jeu (POST) -  fournit le type de jeu
 	apiRouter.POST("/games", server.createNewGameHandler) // retourne un id
+	// creation du jeu (POST) -  fournit le type de jeu
```
Comment (ggerbaud) [inline server/server.go]: You should not copy/paste ;)
Wrong comment

File: webapp/app/scripts/controllers/game.js [hunk 3]
```diff
@@ -71,33 +71,22 @@ angular.module('gdApp')
...
+   
+    // join websocket
+    dataService
+        .joinGame($routeParams.id)
         .then(
-          function (game) {
-            $scope.game = game;
-            timeout();
+            function (ws) {
+            console.log("got a fucking ws to bind")
```
Comment (ggerbaud) [inline webapp/app/scripts/controllers/game.js]: hum...

**Comment by mathieu-pousse** [comment]
fixed all comments

**Comment by ggerbaud** [comment]
LGTM

**Comment by erwannT** [comment]
LGTM

**Pull request merged**